src/coreclr/src/System.Private.CoreLib/src/System/Threading/Thread.CoreCLR.cs

```diff
@@ -486,50 +486,209 @@ private static int CalculateOptimalMaxSpinWaitsPerSpinIteration()
         [MethodImpl(MethodImplOptions.InternalCall)]
         private static extern int GetCurrentProcessorNumber();
 
-        // The upper bits of t_currentProcessorIdCache are the currentProcessorId. The lower bits of
-        // the t_currentProcessorIdCache are counting down to get it periodically refreshed.
-        // TODO: Consider flushing the currentProcessorIdCache on Wait operations or similar
-        // actions that are likely to result in changing the executing core
-        [ThreadStatic]
-        private static int t_currentProcessorIdCache;
-
-        private const int ProcessorIdCacheShift = 16;
-        private const int ProcessorIdCacheCountDownMask = (1 << ProcessorIdCacheShift) - 1;
-        private const int ProcessorIdRefreshRate = 5000;
-
-        private static int RefreshCurrentProcessorId()
+        private static class CoreIdCache
         {
-            int currentProcessorId = GetCurrentProcessorNumber();
+            // The upper bits of t_currentProcessorIdCache are the currentProcessorId. The lower bits of
```

Comment:
This is non-trivial amount of code. It may look better if it is moved to a separate file.